enhancements/baremetal/continuous-hw-introspection.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Continuous HW introspecton of Baremetal IPI hosts
+authors:
+  - "@sadasu"
+reviewers:
+  - "@dhellmann"
+  - "@enxebre"
+  - "@hardys"
+  - "@stbenjam"
+  - "@juliakreger"
+approvers:
+  - "@dhellmann"
+  - "@enxebre"
+creation-date: 2020-02-18
+last-updated: 2020-02-17
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+---
+
+# Continuous HW introspection of Baremetal IPI hosts
+
+This proposal talks about how performing continuous HW introspection on all master and
+and worker nodes in a Baremetal IPI deployment would get us important data about the
+baremetal hosts which potentially can be used to monitor the health of individual nodes and
+the deployment as a whole.
+
+This enhancement will not result in a new CRD or API to be written but will detail
+data that would become availabe in the Status field of the BareMetalHost CRD.
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Continuous HW inspection on all the master and worker nodes that are a part of
+a Baremetal IPI deployment, allow the user to detect changes in server hardware
+and become aware of any alarms or failures seen on the servers. This information
```

Comment:
Does the inspection client actually collect alarms or failures? I thought it just looked at what hardware was present in the server. https://github.com/metal3-io/metal3-docs/blob/master/design/hardware-status.md talks about SMART for drive stats, but I thought that was added to the node_exporter data separately.

Reply:
Inspection client does not actually collect alarms. I was looking at these fields in the baremetal host CRD : https://github.com/metal3-io/baremetal-operator/blob/master/deploy/crds/metal3.io_baremetalhosts_crd.yaml#L211-L222. Based on the type of error, module that is reading this field could potentially raise an alarm.

Reply:
OK, those fields aren't related to inspection, they're general status fields reported by the baremetal-operator as it reconciles the host.